server/src/test/java/org/opentosca/toscana/plugins/cloudfoundry/CloudFoundryPluginTest.java

```diff
@@ -0,0 +1,102 @@
+package org.opentosca.toscana.plugins.cloudfoundry;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Set;
+
+import org.opentosca.toscana.core.BaseUnitTest;
+import org.opentosca.toscana.core.plugin.PluginFileAccess;
+import org.opentosca.toscana.core.transformation.logging.Log;
+import org.opentosca.toscana.model.EffectiveModel;
+import org.opentosca.toscana.model.node.RootNode;
+import org.opentosca.toscana.model.visitor.VisitableNode;
+import org.opentosca.toscana.plugins.cloudfoundry.visitors.CloudFoundryNodeVisitor;
+import org.opentosca.toscana.plugins.testdata.TestEffectiveModels;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.slf4j.LoggerFactory;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+public class CloudFoundryPluginTest extends BaseUnitTest {
+
+    private final static EffectiveModel lamp = new TestEffectiveModels().getLampModel();
+    private static CloudFoundryApplication myApp = new CloudFoundryApplication();
+    private final static CloudFoundryNodeVisitor visitor = new CloudFoundryNodeVisitor(myApp);
+    @Mock
+    private Log log;
+    private File targetDir;
+    private final String appName = "my_app";
+    private final ArrayList<String> paths = new ArrayList<>();
+
+    @Before
+    public void setUp() throws Exception {
+        File sourceDir = new File("src/test/resources/csars/yaml/valid/lamp-input/");
+        targetDir = new File(tmpdir, "targetDir");
+        sourceDir.mkdir();
+        targetDir.mkdir();
+        when(log.getLogger(any(Class.class))).thenReturn(LoggerFactory.getLogger("Test logger"));
+        PluginFileAccess fileAccess = new PluginFileAccess(sourceDir, targetDir, log);
+        Set<RootNode> nodes = lamp.getNodes();
+
+        paths.add("my_app/myphpapp.php");
+        paths.add("my_app/mysql-credentials.php");
+        paths.add("my_app/create_myphpapp.sh");
+        paths.add("my_app/configure_myphpapp.sh");
+        paths.add("mysql_dbms/mysql_dbms_configure.sh");
+
+        for (VisitableNode node : nodes
+            ) {
+            node.accept(visitor);
+        }
+        myApp = visitor.getFilledApp();
+        CloudFoundryFileCreator fileCreator = new CloudFoundryFileCreator(fileAccess, myApp);
+        fileCreator.createFiles();
+    }
+
+    @Test
+    public void getName() {
+        assertEquals(appName, myApp.getAppName());
+    }
+
+    @Test
+    public void getManifest() throws Exception {
+        File targetFile = new File(targetDir, CloudFoundryFileCreator.MANIFEST);
+        String manifestContent = FileUtils.readFileToString(targetFile);
+
+        File expectedFile = new File("src/test/resources/cloudFoundry/lamp-manifest.yml");
```

Comment:
using this long path the second time, hardcoded -- time for introducing a field!